components/brave_wallet/browser/internal/hd_key_ed25519.cc

```diff
@@ -5,55 +5,83 @@
 
 #include "brave/components/brave_wallet/browser/internal/hd_key_ed25519.h"
 
+#include <memory>
 #include <utility>
 
 #include "base/check.h"
-#include "base/logging.h"
-#include "base/notreached.h"
-#include "base/strings/strcat.h"
+#include "base/containers/span.h"
+#include "base/containers/span_writer.h"
+#include "base/containers/to_vector.h"
+#include "base/numerics/byte_conversions.h"
 #include "base/strings/string_number_conversions.h"
 #include "brave/third_party/bitcoin-core/src/src/base58.h"
+#include "crypto/hmac.h"
+#include "third_party/boringssl/src/include/openssl/curve25519.h"
 
 namespace brave_wallet {
 namespace {
-constexpr char kMasterNode[] = "m";
+
+inline constexpr char kMasterSecret[] = "ed25519 seed";
+inline constexpr uint32_t kHardenedOffset = 0x80000000;
+
+// Returns concatenation of (private key, pubic key)
+std::array<uint8_t, ED25519_PRIVATE_KEY_LEN> MakeKeyPair(
+    base::span<const uint8_t> private_key,
+    base::span<const uint8_t> public_key) {
+  std::array<uint8_t, ED25519_PRIVATE_KEY_LEN> key_pair = {};
+  base::span(key_pair).first<kEd25519PrivateKeySize>().copy_from(private_key);
+  base::span(key_pair).last<kEd25519PublicKeySize>().copy_from(public_key);
```

Comment:
We have `ED25519_PRIVATE_KEY_LEN`, but we also have `kEd25519PrivateKeySize`. The distinction between these two doesn't seem very clear to me, as they have identical names.

Reply:
I've changed underlying storage to key pair, added some static_asserts and comments. Hope it's more clear now.